### PR TITLE
2fa toggle bug fix

### PR DESCRIPTION
**Pull request by ashilkn**

### CODE REVIEW

File: lib/core/configuration.dart [hunk 1]
```diff
@@ -68,6 +69,8 @@ class Configuration {
   static const keyShouldShowLockScreen = "should_show_lock_screen";
   static const keyHasSelectedAnyBackupFolder =
       "has_selected_any_folder_for_backup";
+  static const keyHasEnabledTwoFactor = "has_enabled_two_factor";
```
Comment (ua741) [inline lib/core/configuration.dart:72]: Should we move this config key and logic insider UserService?  Configuration is already doing too many things which it should not be doing :)

File: lib/events/two_factor_status_change_event.dart [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-
+import 'package:photos/core/configuration.dart';
 import 'package:photos/events/event.dart';
 
 class TwoFactorStatusChangeEvent extends Event {
   final bool status;
 
-  TwoFactorStatusChangeEvent(this.status);
+  TwoFactorStatusChangeEvent(this.status) {
+    Configuration.instance.setTwoFactor(value: status);
```
Comment (ua741) [inline lib/events/two_factor_status_change_event.dart:8]: This doesn't seem like right place to do this :). Inside UserService,  when the call to enable or disable 2FA is completed, we can set this flag?

File: lib/services/user_service.dart [hunk 3]
```diff
@@ -609,7 +609,7 @@ class UserService {
     }
   }
 
-  Future<void> setupTwoFactor(BuildContext context) async {
+  Future<void> setupTwoFactor(BuildContext context, Completer completer) async {
```
Comment (ua741) [inline lib/services/user_service.dart:612]: Passing completer here seems weird. Let's discuss if we really need this.

File: lib/events/two_factor_status_change_event.dart (hunk 2 above)
Comment (ashilkn) [reply lib/events/two_factor_status_change_event.dart:8]: Did this in a slightly different way. Attached a listener to to the `TwoFactorStatusChangeEvent` which will set the two factor status flag when the event is fired.

File: lib/services/user_service.dart (hunk 3 above)
Comment (ashilkn) [reply lib/services/user_service.dart:621]: Will remove this in the future when the toggles are moved to a separate screen

File: lib/services/user_service.dart [hunk 4]
```diff
@@ -49,6 +53,11 @@ class UserService {
   Future<void> init() async {
     emailValueNotifier =
         ValueNotifier<String>(Configuration.instance.getEmail());
+    _preferences = await SharedPreferences.getInstance();
+    setTwoFactor(fetchTwoFactorStatus: true);
```
Comment (ua741) [inline lib/services/user_service.dart:57]: Add .ignore?

**Review by ua741 — APPROVED** [review]

**Pull request merged**